text/0628-prettier.md

```diff
@@ -0,0 +1,140 @@
+}
+```
+
+#### Prettier
+
+A new `prettier` section will be added to the `package.json` file with the following contents:
+
+```json
+{
+  "prettier": {
+    "singleQuote": true
+  }
+}
+```
+
+There is a [small number of configuration
+options](https://prettier.io/docs/en/options.html) that can be used in this
+file, but we are intentionally avoiding modifying the default values for things
+that are not _nearly_ universally agreed on in the Ember ecosystem.
+
+### Blueprint Changes
+
+In general, it is recommended that all blueprints provided by addons should
+satisfy the default linting configuration of a new Ember application. As such
+the blueprints provided by `ember-cli`, `ember-source`, and `ember-data` will
+be updated to ensure that they satisfy these new linting rules requirements.
+
+#### `package.json` scripts
+
+The `app` and `addon` blueprints will be updated to add the following
+additional entries to `scripts`:
+
+```json
+{
+  "scripts": {
+    "lint": "npm-run-all --aggregate-output --continue-on-error --parallel 'lint:!(fix)'",
+    "lint:fix": "npm-run-all --aggregate-output --continue-on-error --parallel lint:*:fix",
+    "lint:js:fix": "eslint . --fix",
+    "lint:hbs:fix": "ember-template-lint . --fix"
+  }
+}
+```
+The `lint:fix`, `lint:js:fix`, `lint:hbs:fix` scripts are new (introduced with
+this RFC), and the `lint` script will be updated to ensure that it avoids
+running the new `lint:fix`.
+
+This configuration is specifically intending to allow users to add additional
+linters (e.g. `stylelint` or `markdownlint`) by adding scripts for them, and
+they would automatically be rolled up into `lint:fix`.
+
+### Codemod
+
+In order to make the migration as simple as possible, a codemod will be created: `ember-cli-prettier-codemod`.
+
+The codemod will do the following:
+
+* Install the new/required dependencies
+* Migrate an application / addon to the new linting configuration
+* Commit _just_ the configuration changes.
+* Run the new `lint:fix` script to update all files to the new format.
+* Commit just the auto-fixed output.
+
+While this seems quite simple (and in fact it is pretty easy), leveraging a
+codemod will make it much easier to deal with rebasing large Prettier migration
+pull requests. If one of those pull requests gets out of date (e.g. due to a
+conflict in one of the files) you can simply re-run the codemod on the branch
+and it will replace the previously created commits.
+
+## How we teach this
+
+We do not currently discuss linting or stylistic formatting in either guides.emberjs.com or cli.emberjs.com.
+
+This RFC proposes adding a new subsection that discusses linting under [`Basic Use` section of the CLI guides](https://cli.emberjs.com/release/basic-use/).
+
+## Drawbacks
+
+The largest drawback is generally the cost of the _initial_ migration to Prettier. That migration tends to be quick, but it can cause pain for folks maintaining long lived branches.
+
+## Alternatives
+
+Do nothing.
```

Comment:
standardjs could be added as an alternative and rejected for the reasons mentioned in the thread, so we have the rationale documented.

Reply:
Ya, great point! Will update.